src/main/iidxhook3/avs-boot.c

```diff
@@ -0,0 +1,178 @@
+#define LOG_MODULE "avs-boot"
+
+#include <stdint.h>
+#include <string.h>
+
+#include "hook/table.h"
+
+#include "imports/avs.h"
+
+#include "iidxhook3/avs-boot.h"
+
+#include "util/log.h"
+
+static void (*real_avs_boot)(
+    struct property_node *config,
+    void *std_heap,
+    size_t sz_std_heap,
+    void *avs_heap,
+    size_t sz_avs_heap,
+    avs_log_writer_t log_writer,
+    void *log_context);
+static int (*real_ea3_boot_avs)(struct property_node *config);
+static int (*real_ea3_boot)(struct property_node *config);
+
+static void my_avs_boot(
+    struct property_node *config,
+    void *std_heap,
+    size_t sz_std_heap,
+    void *avs_heap,
+    size_t sz_avs_heap,
+    avs_log_writer_t log_writer,
+    void *log_context);
+static int my_ea3_boot_avs(struct property_node *config);
+static int my_ea3_boot(struct property_node *config);
+
+static struct net_addr iidxhook3_avs_boot_eamuse_server_addr;
+
+static const struct hook_symbol iidxhook3_avs_hook_syms[] = {
+    {.name = "avs_boot",
+     .patch = my_avs_boot,
+     .link = (void **) &real_avs_boot},
+    {.name = "ea3_boot",
+     .patch = my_ea3_boot_avs,
+     .link = (void **) &real_ea3_boot_avs},
+};
+
+static const struct hook_symbol iidxhook3_ea3_hook_syms[] = {
+    {.name = "ea3_boot",
+     .patch = my_ea3_boot,
+     .link = (void **) &real_ea3_boot},
+};
```

Comment:
question: Why are there two`ea3_boot` functions?

Reply:
It's because the implementation of the `ea3_boot` function varies by game version.

https://github.com/gar6a9e/bemanitools/blob/legacy-iidx-network/src/main/iidxhook3/avs-boot.c#L148